baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/RawTypes.java

```diff
@@ -0,0 +1,104 @@
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.ClassTree;
+import com.sun.source.tree.MethodTree;
+import com.sun.source.tree.NewClassTree;
+import com.sun.source.tree.ParameterizedTypeTree;
+import com.sun.source.tree.Tree;
+import com.sun.source.tree.VariableTree;
+import com.sun.tools.javac.code.Type;
+import java.util.ArrayList;
+import java.util.List;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        name = "RawTypes",
+        link = "https://github.com/palantir/gradle-baseline#baseline-error-prone-checks",
+        linkType = BugPattern.LinkType.CUSTOM,
+        severity = BugPattern.SeverityLevel.WARNING,
+        // Support SuppressWarnings("rawtypes"), which is already in use
+        altNames = {"rawtypes"},
+        summary =
+                "Avoid raw types; add appropriate type parameters if possible.\n"
+                        + "This can be suppressed with @SuppressWarnings(\"rawtypes\") where necessary, such as when "
+                        + "interacting with older library code.")
+public final class RawTypes extends BugChecker
+        implements BugChecker.VariableTreeMatcher,
+                BugChecker.NewClassTreeMatcher,
+                BugChecker.ClassTreeMatcher,
+                BugChecker.MethodTreeMatcher {
+    @Override
+    public Description matchVariable(VariableTree tree, VisitorState state) {
+        return testType(tree.getType());
+    }
+
+    @Override
+    public Description matchNewClass(NewClassTree tree, VisitorState state) {
+        return testType(tree.getIdentifier());
+    }
+
+    @Override
+    public Description matchClass(ClassTree tree, VisitorState state) {
+        List<Tree> types = new ArrayList<>();
+        types.add(tree.getExtendsClause());
+        types.addAll(tree.getImplementsClause());
+        return testTypes(types);
+    }
+
+    @Override
+    public Description matchMethod(MethodTree tree, VisitorState state) {
+        return testType(tree.getReturnType());
+    }
+
+    private Description testTypes(Iterable<? extends Tree> types) {
+        for (Tree type : types) {
+            Description description = testType(type);
+            if (description != Description.NO_MATCH) {
+                return description;
+            }
+        }
+        return Description.NO_MATCH;
+    }
+
+    private Description testType(Tree type) {
+        if (type == null) {
+            return Description.NO_MATCH;
+        }
+        Type realType = ASTHelpers.getType(type);
+        if (realType != null && realType.isRaw()) {
+            return buildDescription(type)
+                    .setMessage("Avoid raw types; add appropriate type parameters if possible. "
+                            + "The type was: "
+                            + realType
```

Comment:
types can render in odd ways, especially when they include other generic type information. prettyType should make it a bit cleaner:
```suggestion
                            + MoreSuggestedFixes.prettyType(null, null, realType)
```